providers/azureblob/src/main/java/org/jclouds/azure/storage/filters/SharedKeyLiteAuthentication.java

```diff
@@ -110,7 +188,42 @@ HttpRequest replaceDateHeader(HttpRequest request) {
       request = request.toBuilder().replaceHeaders(Multimaps.forMap(builder.build())).build();
       return request;
    }
+   
+	/**
+	* this is the method to parse container name and blob name from the HttpRequest. 
+	* applicable for the cases with SAS Authentication 
+	* @param String uri - result of HttpRequest.getRequestLine();
+	* 
+	*/ 
+   public String[] cutUri (String uri) {
+	    
+		String[] result = new String[2];
+		try {
+			//split the url at "?" point:
+			result = uri.split(" ");
+			
+			//now we need the left part only: 
+			String centrePart = result[1]; 
 
+			//now we need the piece, situated to the right of ".net/":
+			result = centrePart.split(".net/");
+			String path = result[1];
+			
+			//finally, separate container and blob names: 
+			result = path.split("/");
```

Comment:
Would `new URI(String)` solve this more cleanly?

Reply:
changed